device-identifier/shbdeviceidentifier/commands.py

```diff
@@ -43,4 +51,38 @@ def read(db: Database, file_path: click.Path, file_type: str):
             else:
                 logger.error(f"Failed to write {file_path} to Database.")
         else:
-            logger.error(f"Failed to read {file_path}.")
\ No newline at end of file
+            logger.error(f"Failed to read {file_path}.")
+
+
+def train(
+    model_name,
+    use_database: bool,
+    training_data_path: str,
+    training_labels_path: str,
+    devices_to_train: List[str] = None,
+):
+    """Train an ML model."""
+    logger.debug(
+        "Will perform training with following parameters:"
+        f"model_name={model_name}, use_database={use_database}, training_data_path={training_data_path}, training_labels_path={training_labels_path}, devices_to_train={devices_to_train}"
+    )
+    if use_database:
+        params = {
+            "_start": datetime.strptime("2022-08-01T11:40:00UTC", "%Y-%m-%dT%H:%M:%S%Z"),
+            "_stop": datetime.strptime("2022-08-01T11:41:00UTC", "%Y-%m-%dT%H:%M:%S%Z"),
+        }
+        query = """
+                    from(bucket: "network-traffic")
+                    |> range(start: _start, stop: _stop)
+                    |> filter(fn: (r) => r["_measurement"] == "packet")  
+                """
+        train_df, train_labels = DataLoader.from_database(query, params, devices_to_train)
+    else:
+        train_df, train_labels = DataLoader.from_file(
+            training_data_path, training_labels_path, devices_to_train
+        )
+    model = get_model(model_name)
+    model.train(train_df[["data_len", "stream_id"]], train_labels)
```

Comment:
We should probably make sure that there is some kind of callback that shows training progress, otherwise using this from the CLI with a longer training process will be problematic. Should open a new issue for this.

Reply:
Sound reasonable. Quick googling indicates that this could become a bit problematic, as training for sklearn differs from training in torch etc. This means that we might have to implement this on the model level.

Reply:
Yes, MLModel should have some kind of progress method.

Reply:
Is there already an issue for this?

Reply:
Opened one: #96.